exec/bigmachine_test.go

```diff
@@ -171,6 +171,129 @@ func TestBigmachineExecutorPanicCompile(t *testing.T) {
 	run(t, x, tasks, TaskErr)
 }
 
+// TestBigmachineExecutorProcs verifies that using the Procs pragma properly
+// affects machine/proc allocation.
+func TestBigmachineExecutorProcs(t *testing.T) {
+	// Set up the test with:
+	// - a slice with 8 tasks
+	// - Procs(2) so that two procs are allocated for each task
+	// - a system with 12 procs, 4 procs per machine
+	system := testsystem.New()
+	system.Machineprocs = 4
+	ctx, cancel := context.WithCancel(context.Background())
+	x := newBigmachineExecutor(system)
+	shutdown := x.Start(&Session{
+		Context: ctx,
+		p:       12, // 3 machines
+		maxLoad: 1,
+	})
+	defer shutdown()
+	defer cancel()
+
+	// We use the wait group to block completion of tasks, controlling execution
+	// for our test.
+	var wg sync.WaitGroup
```

Comment:
Optional: I found use of `WaitGroup` to be confusing at first because I expected we'd be counting completed tasks, etc. How about waiting for a channel to close instead? I think that's idiomatic.